pkg/authorization/api/conversion.go

```diff
@@ -0,0 +1,192 @@
+		Convert_rbac_Role_To_api_Role,
+		Convert_rbac_ClusterRoleBinding_To_api_ClusterRoleBinding,
+		Convert_rbac_RoleBinding_To_api_RoleBinding,
+	); err != nil { // If one of the conversion functions is malformed, detect it immediately.
+		return err
+	}
+	return nil
+}
+
+func Convert_api_ClusterRole_To_rbac_ClusterRole(in *ClusterRole, out *rbac.ClusterRole, _ conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Rules = convertOriginPolicyRule(in.Rules)
+	return nil
+}
+
+func Convert_api_Role_To_rbac_Role(in *Role, out *rbac.Role, _ conversion.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Rules = convertOriginPolicyRule(in.Rules)
+	return nil
+}
+
+func Convert_api_ClusterRoleBinding_To_rbac_ClusterRoleBinding(in *ClusterRoleBinding, out *rbac.ClusterRoleBinding, _ conversion.Scope) error {
+	var err error
+	if out.Subjects, err = convertOriginSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertOriginRoleRef(&in.RoleRef)
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func Convert_api_RoleBinding_To_rbac_RoleBinding(in *RoleBinding, out *rbac.RoleBinding, _ conversion.Scope) error {
+	if len(in.RoleRef.Namespace) != 0 && in.RoleRef.Namespace != in.Namespace {
+		return fmt.Errorf("invalid origin role binding %s: attempts to reference role in namespace %q instead of current namespace %q", in.Name, in.RoleRef.Namespace, in.Namespace)
+	}
+	var err error
+	if out.Subjects, err = convertOriginSubjects(in.Subjects); err != nil {
+		return err
+	}
+	out.RoleRef = convertOriginRoleRef(&in.RoleRef)
+	out.ObjectMeta = in.ObjectMeta
+	return nil
+}
+
+func convertOriginPolicyRule(in []PolicyRule) []rbac.PolicyRule {
+	rules := make([]rbac.PolicyRule, 0, len(in))
+	for _, rule := range in {
+		r := rbac.PolicyRule{ // AttributeRestrictions is lost, but our authorizor ignores that field now
+			APIGroups:       rule.APIGroups,
+			Verbs:           rule.Verbs.List(),
+			Resources:       rule.Resources.List(),
+			ResourceNames:   rule.ResourceNames.List(),
+			NonResourceURLs: rule.NonResourceURLs.List(),
+		}
+		rules = append(rules, r)
+	}
+	return rules
+}
+
+func convertOriginSubjects(in []api.ObjectReference) ([]rbac.Subject, error) {
+	subjects := make([]rbac.Subject, 0, len(in))
+	for _, subject := range in {
+		s := rbac.Subject{
+			Name:       subject.Name,
+			APIVersion: rbac.GroupName, // TODO what to use here?
+		}
+
+		switch subject.Kind {
+		case ServiceAccountKind:
+			s.Kind = rbac.ServiceAccountKind
+			s.Namespace = subject.Namespace
+		case UserKind, SystemUserKind:
+			s.Kind = rbac.UserKind
+		case GroupKind, SystemGroupKind:
+			s.Kind = rbac.GroupKind
+		default:
+			return nil, fmt.Errorf("invalid kind for origin subject: %q", subject.Kind)
+		}
+
+		subjects = append(subjects, s)
+	}
+	return subjects, nil
+}
+
+func convertOriginRoleRef(in *api.ObjectReference) rbac.RoleRef {
+	return rbac.RoleRef{
+		APIGroup: in.APIVersion,
+		Kind:     in.Kind, // TODO leave empty?
```

Comment:
When the origin roleref namespace is empty, the rbacRoleRef kind should be "ClusterRole".  Otherwise, it should be "Role"